### ISSUE TITLE
Simplify configuration

### ISSUE THREAD

**Issue by knsv** [mermaid-js/mermaid#1172]
**Configuration handling**
The configuration handling internally in mermaid is somewhat unstructured. 

* The default values are defined in mermaidApi.
* mermaidApi injects the default values into the config module.
* Then in render the render function injects config into the renderer.

This is overly complex and can be hard to follow.

Instead:
* Move default values from mermaidApi into the config module
* Make mermaid api send the config from the user into the config module where it overrides the default values
* Move any additional configuration in the diagrams into the config modules
* Make all diagrams get all the config data fromt he config module via some simple and short syntax.

**Comment by chrismoran-bkt** [comment]
What do you think about each renderer registering interest in config model changes for their respective channels? 

We could use simple notifications / watches on config.<channel> and renderers could add a listener (watch) on config.<their channel>. For example the sequenceHandler could use this by listening on config.addListener('sequence'). When config.sequence changes, the callback would be invoke for sequenceRenderer. 

With PR #1482 this should be much simpler. [assignWithDepth](https://github.com/mermaid-js/mermaid/blob/24ed979faf7c583624c0ed6f97cac7659f57c701/src/utils.js#L425) enhances Object.assign to be able to merge objects of arbitrary depth (configurable with a default of 2 - which happens to be what mermaid uses/used to effectively use for the previous setConfig functionality :) ). This can be leveraged to also include notifications to interested parties when properties change.

**Comment by jgreywolf** [comment]
@knsv  @chrismoran-bkt Does this issue still need a solution?

**Comment by jgreywolf** [comment]
@sidharthv96 can this be closed similar to #3397

**Comment by sidharthv96** [comment]
I don't think they are similar enough, although implementing #3397 properly would solve this one too. They are somewhat "sibling" issues. 

As a solution for this issue, we could come up with a pattern which we can implement when migrating new diagrams. 
cc @Yokozuna59
